journalpump/daemon.py

```diff
@@ -40,7 +40,7 @@ def __init__(
         self.log = logging.getLogger(self.name)
         self.config_path = config_path
         self.config_file_ctime = None
-        self.config = None
+        self.config = {}
```

Comment:
This gets initialized in the next two lines, setting here so typing checks don't complain, and its obvious that this is always a dict.

The require_config being something that can be disabled also probably doesn't make sense, as lots of code has required keys from a brief grep, but that can be a future change.